zebra/zebra_rib.c

```diff
@@ -457,9 +469,19 @@ nexthop_active_ipv4 (struct rib *rib, struct nexthop *nexthop, int set,
 	    {
 	      /* Directly point connected route. */
 	      newhop = match->nexthop;
-	      if (newhop && nexthop->type == NEXTHOP_TYPE_IPV4)
-		nexthop->ifindex = newhop->ifindex;
-	      
+	      if (newhop)
+		{
+		  if (nexthop->type == NEXTHOP_TYPE_IPV4)
+		    {
+		      nexthop->ifindex = newhop->ifindex;
+		      nexthop->type = NEXTHOP_TYPE_IPV4;
+		    }
+		  if (nexthop->type == NEXTHOP_TYPE_IPV6)
+		    {
+		      nexthop->ifindex = newhop->ifindex;
+		      nexthop->type = NEXTHOP_TYPE_IPV6_IFINDEX;
+		    }
+		}
```

Comment:
As discussed on slack:
I am attempting to auto-convert NEXTHOP_TYPE_IPV[4|6] to NEXTHOP_TYPE_IPV[4|6]_IFINDEX and only store _IFINDEX TYPES in side of zebra.  We have issues in the code where we are doing the wrong thing( by sending wrong data to the kernel ).